WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerViewHolder.kt

```diff
@@ -1,15 +1,40 @@
 package org.wordpress.android.ui.sitecreation.theme
 
-import android.view.LayoutInflater
+import android.graphics.drawable.Drawable
+import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.ImageView.ScaleType.FIT_CENTER
 import androidx.recyclerview.widget.RecyclerView
-import org.wordpress.android.R
-
-class HomePagePickerViewHolder(internal val parent: ViewGroup) :
-        RecyclerView.ViewHolder(
-                LayoutInflater.from(parent.context).inflate(
-                        R.layout.home_page_picker_item,
-                        parent,
-                        false
-                )
-        )
+import kotlinx.android.synthetic.main.home_page_picker_item.view.*
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.image.ImageManager.RequestListener
+import org.wordpress.android.util.image.ImageType
+import org.wordpress.android.util.setVisible
+
+class HomePagePickerViewHolder(view: View, val parent: ViewGroup) : RecyclerView.ViewHolder(view) {
+    private val container: View = itemView.layout_container
+    private val preview: ImageView = itemView.preview
+    private val selected: ImageView = itemView.selected_overlay
+
+    fun onBind(
+        uiState: LayoutGridItemUiState,
+        imageManager: ImageManager
+    ) {
+        imageManager.loadWithResultListener(preview, ImageType.THEME, uiState.preview, FIT_CENTER, null,
+                object : RequestListener<Drawable> {
+                    override fun onLoadFailed(e: Exception?, model: Any?) {
+                    }
+
+                    override fun onResourceReady(resource: Drawable, model: Any?) {
+                        uiState.onThumbnailReady.invoke()
```

Comment:
Thank you for taking the time to review this @mkevins 🙇 

> Not suggesting a change here, just curious: why use invoke() instead of directly calling like: uiState.onThumbnailReady()?

Good catch 👍 
There is no reason for this. I originally used a nullable `onThumbnailReady` type. After removing the nullability I didn't clean up properly. Fixed with bbe22199689cee2cb06a35cf083dd427145dce45

Reply:
Ah, that makes sense. Thanks for the explanation!